From d26b5babf129e3698932c2dc90a2d245d34a9530 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 02 Mar 2023 16:20:52 +0200
Subject: [PATCH] [SW-99614] gaudi2: security: refactor prot bits func

Simplify the init_prot_bits_block() function API:
* remove the obscure boolean parameter - use structure pointer instead
* remove the redundant array index

This straight forward change removes 16 bytes w/o affecting the
functionality and even makes the init_prot_bits_block() function more
generic.

Change-Id: I266ee1fb62b570cdfe2bef62e1378bb19c44ffb0
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index cb9a96c..d7c13f2 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -790,17 +790,11 @@
 		       PCIE_WRAP_PCIE_PROT_OVR_WR_EN_MASK), mmPCIE_WRAP_PCIE_PROT_OVR);
 }
 
-static int init_prot_bits_block(u32 index, u8 major, u8 minor, u8 sub_minor, bool is_reset_conf)
+static int init_prot_bits_block(u8 major, u8 minor, u8 sub_minor, struct hl_prot_bits_cfg *cfg)
 {
-	struct hl_prot_bits_cfg *cfg;
 	u8 b_idx = 0, d_idx = 0;
 	u32 val;
 	u32 addr, reg_addr;
-
-	if (is_reset_conf)
-		cfg = &gaudi2_prot_bits_reset[index];
-	else
-		cfg = &gaudi2_prot_bits_cfg[index];
 
 	addr = cfg->addr.base_addr + HL_GLBL_PRIV_REG_OFFSET + major * cfg->addr.major_offset +
 			  minor * cfg->addr.minor_offset + sub_minor * cfg->addr.sub_minor_offset;
@@ -842,8 +836,8 @@
 			do {
 				sub_minor_idx = 0;
 				do {
-					init_prot_bits_block(cfg_idx, major_idx, minor_idx,
-								sub_minor_idx, false);
+					init_prot_bits_block(major_idx, minor_idx, sub_minor_idx,
+							     &gaudi2_prot_bits_cfg[cfg_idx]);
 					sub_minor_idx++;
 				} while (sub_minor_idx < block_addr.sub_minor);
 				minor_idx++;
@@ -1255,7 +1249,7 @@
 	 * security init are not getting reset even if security is disabled
 	 */
 	for (i = 0; i < ARRAY_SIZE(gaudi2_prot_bits_reset); i++)
-		init_prot_bits_block(i, 0, 0, 0, true);
+		init_prot_bits_block(0, 0, 0, &gaudi2_prot_bits_reset[i]);
 }
 
 #define PROT_OVR_SECURE		0
